components/collector/src/source_collectors/jenkins/pipeline_duration.py

```diff
@@ -0,0 +1,73 @@
+"""Jenkins CI-pipeline duration collector."""
+
+from collections.abc import Sequence
+from datetime import timedelta
+from typing import TypedDict, cast
+
+from base_collectors import SourceCollector
+from collector_utilities.date_time import minutes
+from collector_utilities.functions import match_string_or_regular_expression
+from collector_utilities.type import URL, Value
+from model import SourceResponses
+
+from .base import Build, Job
+
+
+class JSON(TypedDict):
+    """Jenkins JSON."""
+
+    builds: Sequence[Build]  # Workflow jobs have builds
+    jobs: Sequence[Job]  # Multibranch pipelines have jobs that have builds
+
+
+class JenkinsPipelineDuration(SourceCollector):
+    """Jenkins CI-pipeline duration collector."""
+
+    async def _api_url(self) -> URL:
+        """Extend to add the job API path and parameters."""
+        url = await super()._api_url()
+        pipeline_job = self._parameter("pipeline")
+        builds = "builds[duration,result,url]"
+        return URL(f"{url}/job/{pipeline_job}/api/json?tree=jobs[name,{builds}],{builds}")
```

Comment:
Is this duplicated `{builds}` to support both regular and multibranch jobs?

Reply:
Yup. I'll add a comment.